apps/src/lib/tools/jsinterpreter/JSInterpreter.js

```diff
@@ -146,9 +146,8 @@ export default class JSInterpreter {
       this.eventQueue = [];
       // Append our mini-runtime after the user's code. This will spin and process
       // callback functions:
-      options.code += '\nwhile (true) { var obj = getCallback(); ' +
-                      'if (obj) { var ret = obj.fn.apply(null, obj.arguments ? obj.arguments : null);' +
-                      'setCallbackRetVal(ret); }}';
+      options.code += '\n;while(true){var __jsCB=getCallback();' +
+                      'if(__jsCB){setCallbackRetVal(__jsCB.fn.apply(null,__jsCB.arguments || null));}}';
```

Comment:
avoiding global namespace conflict with `obj` from student code. Eliminated the use of the global `ret` variable.